azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/DeploymentTemplate.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.arm.domain;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.jclouds.javax.annotation.Nullable;
+
+import java.util.List;
+import java.util.Map;
+
+@AutoValue
+public abstract class DeploymentTemplate {
+
+   //Empty placeholders as we want to generate the empty JSON object
+   @AutoValue
+   public abstract static class Parameters {
+      public static Parameters create() {
+         return new AutoValue_DeploymentTemplate_Parameters();
+      }
+   }
+
+   public abstract String $schema();
```

Comment:
Yes, I know, but having special characters as part of variable names is not very Java. I suggested to name the property "in Java" just `schema` and configure the `@SerializedNames` to be `$schema`, so you can use the right json value while keeping a clean naming in the Java model.